esx_service/vmci/vmci_client.c

```diff
@@ -360,7 +360,7 @@ Vmci_GetReply(int port, const char* json_request, const char* be_name,
       return CONN_FAILURE;
    }
 
-   req.mlen = strnlen(json_request, MAXBUF) + 1;
+   req.mlen = strnlen(json_request, MAXBUF-1) + 1;
```

Comment:
Sorry missed updating, the original code was creating a string which was MAXBUF + 1, the strnlen() can create a string that can be at most MAXBUF length in size and over that it adds the NULL byte. The change is to create a string thats MAXBUF -1 (1MB -1) plus the NULL byte which makes a total of MAXBUF. We may never have a request size of 1MB but the code is not correct as is.